### DISCUSSION TITLE
qtile bar center

### DISCUSSION THREAD

**Discussion by ghost** [qtile/qtile#3069] [Ideas]
please how ı center this bar 
![2021-12-05-084810_1280x59_scrot](https://user-images.githubusercontent.com/80429360/144735440-af824cb8-2c9c-4234-ad9d-35471189e187.png)

**Comment by m-col** [comment]
Do you mean the text box widget in the middle? If so, you can add some 'spacer' widgets that will fill the space and position the other widgets. First pass your left-hand widgets to your `Bar`, then `widget.Spacer()`, then the center widget(s), then another `widget.Spacer()` and then the widgets for the right hand side.

**Reply by ghost** [reply]
no bro ı'am no say spacer look this image bruh
![2021-12-06-023212_66x58_scrot](https://user-images.githubusercontent.com/80429360/144768471-f3e69c7f-9fb7-4c5a-a402-755d0b4133c6.png) is this normal ? this is not center

**Reply by ghost** [reply]
![2021-12-06-023356_1280x1012_scrot](https://user-images.githubusercontent.com/80429360/144768522-ea2da85d-49e6-49f2-83fb-cfc8dad12f53.png)

**Reply by m-col** [reply]
Ah I see the vertical positioning. It looks like your font size is larger than the height of the bar, which will cause problems. What is the bar's height and the widget's font size?

**Reply by elParaguayo** [reply]
Maybe I need to reopen the PR for vertical padding on text boxes again...

**Reply by ghost** [reply]
ı change widget size but not center vertical 35 bar 35 text size ı change text size 25 this is not center
![2021-12-07-051251_1280x1012_scrot](https://user-images.githubusercontent.com/80429360/144953582-eb0dbb3a-3a62-4adf-8dea-5e602fd55794.png)
![2021-12-07-051255_165x65_scrot](https://user-images.githubusercontent.com/80429360/144953586-38cff5cc-1108-4a56-9521-ceb9d76eb6ed.png)

**Reply by elParaguayo** [reply]
I'm not sure there's a perfect fix for this other than doing something like #2346.

**Reply by ghost** [reply]
![2021-12-07-114257_1280x1012_scrot](https://user-images.githubusercontent.com/80429360/144995827-0dc9da01-f397-4956-b842-977f8ce69639.png)

**Reply by elParaguayo** [reply]
The PR hasn't been merged so `padding_x` and `padding_y` won't do anything to your TextBoxes.

**Reply by ghost** [reply]
how ı use pr ?

**Reply by elParaguayo** [reply]
http://docs.qtile.org/en/latest/manual/howto/git.html#i-want-to-try-out-a-feature-somebody-is-working-on

**Reply by ghost** [reply]
Using git what ? I don't understand, I'm so confused my english not good ı dont see pr example ?

git remote add <user> https://www.github.com/user/qtile  ?????
git fetch upstream pull/<id>/head:pr<id>
git checkout pr<id>
why you send this

**Reply by elParaguayo** [reply]
You need to clone the qtile git repository. You can then use that link to help you download the PR I posted in an earlier comment (#2346) and try that.

If you don't know how to run from the cloned repository, you should also read this page from the docs: https://docs.qtile.org/en/latest/manual/hacking.html

Alternatively, you may just want to wait for a fix to be implemented.

**Reply by ghost** [reply]
I'am not python dev ı dont know what you say but thanks

**Reply by elParaguayo** [reply]
That's fine. If you don't feel comfortable trying you've still helped us by reporting the issue. Thank you.
